tests/integration/devfile/cmd_devfile_log_test.go

```diff
@@ -0,0 +1,104 @@
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+
+		// Devfile push requires experimental mode to be set
+		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+		originalKubeconfig = os.Getenv("KUBECONFIG")
+		helper.LocalKubeconfigSet(context)
+		namespace = cliRunner.CreateRandNamespaceProject()
+		currentWorkingDirectory = helper.Getwd()
+		cmpName = helper.RandString(6)
+		helper.Chdir(context)
+		projectDirPath = context + projectDir
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		cliRunner.DeleteNamespaceProject(namespace)
+		helper.Chdir(currentWorkingDirectory)
+		err := os.Setenv("KUBECONFIG", originalKubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("Verify odo log for devfile works", func() {
+
+		It("should log run command output", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+
+			helper.CmdShouldPass("odo", "push", "--project", namespace)
+			output := helper.CmdShouldPass("odo", "log")
+			Expect(output).To(ContainSubstring("ODO_COMMAND_RUN"))
+
+			// test with follow flag
+			output = helper.CmdShouldRunWithTimeout(1*time.Second, "odo", "log", "-f")
+			Expect(output).To(ContainSubstring("program=devrun"))
+
+		})
+
+		It("should error out if component does not exist", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CmdShouldFail("odo", "log")
+		})
+
+		It("should log debug command output", func() {
+
+			helper.CmdShouldPass("git", "clone", "https://github.com/che-samples/web-nodejs-sample.git", projectDirPath)
+			helper.Chdir(projectDirPath)
+
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), projectDirPath)
+			helper.RenameFile("devfile-with-debugrun.yaml", "devfile.yaml")
+			helper.CmdShouldPass("odo", "push", "--debug")
+
+			output := helper.CmdShouldPass("odo", "log", "--debug")
+			Expect(output).To(ContainSubstring("ODO_COMMAND_DEBUG"))
+
+			// test with follow flag
+			output = helper.CmdShouldRunWithTimeout(1*time.Second, "odo", "log", "-f")
+			Expect(output).To(ContainSubstring("program=debugrun"))
+
+		})
+
+		// we do not need test for run command as odo push fails
+		// if there is no run command in devfile.
+		It("should give error if no debug command in devfile", func() {
+
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+
+			helper.CmdShouldPass("odo", "push", "--project", namespace)
+			helper.CmdShouldFail("odo", "log", "--debug")
```

Comment:
It seems like this test is failing
https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/3384/pull-ci-openshift-odo-master-v4.4-integration-e2e/1280783116180393984#1:build-log.txt%3A605

Reply:
/hold

Reply:
Updated, failing as all devfiles in registry now have debug command

Reply:
Still failing https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/3384/pull-ci-openshift-odo-master-v4.4-integration-e2e/1280783116180393984#1:build-log.txt%3A603 I guess it is still pulling the devfile from the registry